mesheryctl/pkg/utils/helpers_test.go

```diff
@@ -0,0 +1,504 @@
+package utils
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"testing"
+)
+
+func getFixturesDirectory() string {
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		log.Fatal("TestBackupConfigFile: Cannot get current working directory")
+	}
+	currDir := filepath.Dir(filename)
+	// get the fixtures file directory
+	fixturesDir := filepath.Join(currDir, "fixtures")
+	return fixturesDir
+}
+
+var fixturesDir = getFixturesDirectory()
+
+func TestBackupConfigFile(t *testing.T) {
+	name := "config.yaml"
+	configFilePath := filepath.Join(fixturesDir, name)
+
+	backupFileName := "config.bak.yaml"
+	backupConfigFilePath := filepath.Join(fixturesDir, backupFileName)
+
+	// creates a config file
+	NewGoldenFile(t, name, fixturesDir).Write("mesheryctl")
+
+	BackupConfigFile(configFilePath)
+
+	// check if backup file is present or not
+	_, err := os.Stat(backupConfigFilePath)
+	if err != nil {
+		t.Errorf("BackupConfigFile error = %v", err)
+	}
+}
+
+func TestStringWithCharset(t *testing.T) {
+	// checking the length, since this function returns random strings everytime
+	strLength := 10
+
+	gotString := StringWithCharset(strLength)
+	actualStringLength := len(gotString)
+
+	if actualStringLength != strLength {
+		t.Errorf("StringWithCharset got = %v want = %v", actualStringLength, strLength)
+	}
+}
+
+func TestPrereq(t *testing.T) {
+	_, _, err := prereq()
+	if err != nil {
+		t.Errorf("prereq error = %v", err)
+	}
+}
+
+func TestSetFileLocation(t *testing.T) {
+	err := SetFileLocation()
+	if err != nil {
+		t.Errorf("SetFileLocation error = %v", err)
+	}
+}
+
+// func TestNavigateToBrowser(t *testing.T) {
+// 	// opens up a browser window whenever this test runs
+// 	err := NavigateToBrowser("https://www.layer5.io")
+// 	if err != nil {
+// 		t.Errorf("NavigateToBrowser error: %v", err)
+// 	}
+// }
+
+func TestUploadFileWithParams(t *testing.T) {
+	fixtureFileName := "listmanifest.api.response.golden" // any arbitrary fixture file
+	uploadFilePath := filepath.Join(fixturesDir, fixtureFileName)
+
+	test := struct {
+		uri       string
+		params    map[string]string
+		paramName string
+		path      string
+	}{
+		uri:       "https://www.layer5.io",
+		params:    nil,
+		paramName: "meshery",
+		path:      uploadFilePath,
+	}
+
+	// returns *http.Request
+	_, err := UploadFileWithParams(test.uri, test.params, test.paramName, test.path)
+
+	if err != nil {
+		t.Errorf("UploadFileWithParams error = %v", err)
+	}
+}
+
+func TestContentTypeIsHTML(t *testing.T) {
+	tests := []struct {
+		name     string
+		response *http.Response
+		want     bool
```

Comment:
what's "want" here?

Reply:
want is be the expected output. Should I rename it to `expectedOutput`?

Reply:
renaming it would be better for someone reading it in futuree.